CONTRIBUTING.md

```diff
@@ -0,0 +1,141 @@
+
+This project only uses *publicly available code*. In particular, do not use any Nintendo source code leaks.
+
+### Environment Setup
+
+Get started by following our [installation guide](https://github.com/zeldaret/mm/wiki/Installation).
+When successful, you should be able to build a matching ROM before you start making any changes.
+
+### First Contribution
+
+Usually, the best place to get started is to decompile an actor overlay. 
+An *actor* is any thing in the game that moves or performs actions or interactions: Link is an actor, enemies are actors, NPCs are actors, etc.
+
+You should [join the Discord](https://discord.zelda64.dev/) to say hello and get suggestions on where to start on the `#mm-decomp` channel.
+
+We track who is working on what on the [Trello board](https://trello.com/b/ruxw9n6m/majoras-mask-decompilation).
+After joining the Discord, ask to be added to the Trello board so you can claim your actor and avoid duplicate work.
+
+Style Guide & Conventions
+-------------------------
+
+Most of the C formatting style is enforced by the `format.sh` script, which is based on `clang-format`.
+Running `./format.sh` will apply our standard style to all `.c` files in the repository.
+
+There are some conventions that cannot be automatically enforced.
+
+### Naming Scheme
+
+(TODO)
+- Naming scheme
+- `ActorFunc` naming specifics
+- Actor `struct` & `enum` defintions go in the header file
+- Renaming symbols referenced by asm (?)
+- Documentation step - what does this entail? (Not required for actors, but if started should be complete. Should be required for code/boot)
+
+### `NON_MATCHING`
+
+If you make significant progress on decompiling a function, but can't get it to match perfectly, you can use a `NON_MATCHING` block to commit your code but have it excluded from the build, like so:
+
+```c
+#ifdef NON_MATCHING
+void CollisionCheck_SpawnWaterDroplets(GlobalContext* ctxt, Vec3f* v) {
+    // ... 
+}
+#else
+#pragma GLOBAL_ASM("./asm/non_matchings/code/z_collision_check/CollisionCheck_SpawnWaterDroplets.asm")
+#endif
+```
+
+### Matching vs. Documenting
+
+Usually, the first step of decompiling a section of code is to get it *matching*: to produce a C version of the code that can be compiled into an identical ROM.
+
+However, the goal of this project is to produce a codebase that can be understood and modified.
+So, beyond producing matching C code, the next steps are *documenting* the code.
+
+Documenting is more than just adding comments, and entails:
+
+- Renaming functions, variables, and struct members
+- Using (or adding) constants and macros when possible
+- Commenting sections of code that are not straightforward
+- (TODO)
+
+It is expected that additions to `boot/` and `code/` are fully documented.
+Overlays do not need to be fully documented, but it is encouraged.
```

Comment:
Overlays are not required to be documented at this time, but code and boot are.
When an overlay documentation has been started, it will be required to be completed.

Reply:
I also want to add that documenting variable names for object segment symbols should not be done yet. These will be given names when object reconstruction is ready. Comments with name suggestions are very much welcome.